flagsmith-core.ts

```diff
@@ -613,21 +633,24 @@ const Flagsmith = class {
 
     getTrait = (key: string) => {
         const trait = this.traits && this.traits[key.toLowerCase().replace(/ /g, '_')];
+        if (isTraitConfig(trait)) {
+            return trait.value;
+        }
         return trait;
     }
 
     getAllTraits = () => {
         return this.traits
     }
 
-    setTrait = (key: string, trait_value: IFlagsmithTrait) => {
+    setTrait = (key: string, trait_value: IFlagsmithTrait, transient?: boolean) => {
```

Comment:
setTraits would need adjusting too

Reply:
See https://github.com/Flagsmith/flagsmith-js-client/pull/240#discussion_r1684737877

Reply:
Ah maybe I'm wrong based on ITraits changing?

Reply:
I believe so — setTraits only updates the storage and the new types get taken care of when they're being sent to API.